src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -306,19 +312,34 @@ JsonNode TransformSchemaNode(JsonSchemaExporterContext ctx, JsonNode schema)
                 ConvertSchemaToObject(ref schema).Insert(0, DescriptionPropertyName, (JsonNode)attr.Description);
             }
 
-            // If the type is recursive, the resulting schema will contain a $ref to the type itself.
-            // As JSON pointer doesn't support relative paths, we need to fix up such paths to accommodate
-            // the fact that they're being nested inside of a higher-level schema.
-            if (schema is JsonObject refObj && refObj.TryGetPropertyValue(RefPropertyName, out JsonNode? paramName))
+            if (schema is JsonObject objSchema)
             {
-                // Fix up any $ref URIs to match the path from the root document.
-                string refUri = paramName!.GetValue<string>();
-                Debug.Assert(refUri is "#" || refUri.StartsWith("#/", StringComparison.Ordinal), $"Expected {nameof(refUri)} to be either # or start with #/, got {refUri}");
-                refUri = refUri == "#"
-                    ? $"#/{PropertiesPropertyName}/{key.ParameterName}"
-                    : $"#/{PropertiesPropertyName}/{key.ParameterName}/{refUri.AsMemory("#/".Length)}";
-
-                refObj[RefPropertyName] = (JsonNode)refUri;
+                // The resulting schema might be a $ref using a pointer to a different location in the document.
+                // As JSON pointer doesn't support relative paths, parameter schemas need to fix up such paths
+                // to accommodate the fact that they're being nested inside of a higher-level schema.
+                if (key.ParameterName is not null && objSchema.TryGetPropertyValue(RefPropertyName, out JsonNode? paramName))
+                {
+                    // Fix up any $ref URIs to match the path from the root document.
+                    string refUri = paramName!.GetValue<string>();
```

Comment:
TryGetPropertyValue doesn't guarantee a non-null node on success?

Reply:
It doesn't because JsonNode uses `null` to represent JSON nulls.